sympy/core/function.py

```diff
@@ -1005,10 +1005,15 @@ def __new__(cls, expr, *variables, **assumptions):
             variables = expr.free_symbols
             if len(variables) != 1:
                 from sympy.utilities.misc import filldedent
-                raise ValueError(filldedent('''
-                    Since there is more than one variable in the
-                    expression, the variable(s) of differentiation
-                    must be supplied to differentiate %s''' % expr))
+                if len(variables) == 0:
```

Comment:
Here's the first line of the docstring for ``diff``

```
>>> (x**2).diff?
Signature: diff(f, *symbols, **kwargs)
Docstring:
Differentiate f with respect to symbols.
...
````

Probably this should be made more clear then?